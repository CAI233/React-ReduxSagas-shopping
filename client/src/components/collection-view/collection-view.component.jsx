import React from 'react';
import { withRouter } from 'react-router-dom';
import { CollectionPreviewStyledContainer, LinkStyled, Preview } from './collection-view.styles';

import CollectionItem from "../collection-item/collection-item.component";

export const CollectionView = ( {title,items,match} ) => {

    const routeUrl = encodeURI(`${match.url}/${title.toLowerCase()}`) ;
    return (
        <CollectionPreviewStyledContainer>
            <LinkStyled to={routeUrl} >
                {title.toUpperCase()} · 查看更多 
            </LinkStyled>
            <Preview>
                {
                    items
                    .filter( (cur,index)=>index<4 ) // 通过索引值限制输出数量
                    .map( ( item )=>( 
                        <CollectionItem key={item.id} item={item} />
                    ) )
                }
            </Preview>
        </CollectionPreviewStyledContainer>
    );

}

export default withRouter(CollectionView);