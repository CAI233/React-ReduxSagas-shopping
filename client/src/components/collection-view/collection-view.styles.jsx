import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { CollectionItemStyledContainer } from '../collection-item/collection-item.styles';

export const CollectionPreviewStyledContainer = styled.div`
    display: flex;
    flex-direction: column;
    margin-bottom: 30px;
`;
CollectionItemStyledContainer.displayName = "CollectionItemStyledContainer";

export const LinkStyled = styled(Link)`
    font-size: 28px;
    margin-bottom: 25px;
`;
LinkStyled.displayName = "LinkStyled";

export const Preview = styled.div`
    display: flex;
    justify-content: space-between;
`;
Preview.displayName = "Preview";

