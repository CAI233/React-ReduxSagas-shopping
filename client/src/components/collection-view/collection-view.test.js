import React from 'react';
import { shallow } from 'enzyme';
import { CollectionView } from './collection-view.component';
import CollectionItem from "../collection-item/collection-item.component";

describe(" 测试单个商品显示 ",()=>{
    let wrapper,mockProps;
    beforeEach(()=>{
        mockProps = {
            title: 'test',
            items: [],
            match: {
                url: '/test'
            }
        }
        wrapper = shallow( <CollectionView { ...mockProps } /> );
    });
    it("CollectionView Snapshot",()=>{
        expect(wrapper).toMatchSnapshot();
    });
    it("验证Item渲染数量",()=>{
        expect(wrapper.find(CollectionItem).length <= 4).toBe(true);
        
    })
});