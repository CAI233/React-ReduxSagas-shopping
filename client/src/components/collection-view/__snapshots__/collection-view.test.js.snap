// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 测试单个商品显示  CollectionView Snapshot 1`] = `ShallowWrapper {}`;
